### PR TITLE
adding javascript addthis_plugin_info variable

### PR DESCRIPTION
**Pull request by julka**
adding javascript addthis_plugin_info variable to help AddThis support people with troubleshooting

### CODE REVIEW

**Comment by matglas** [comment]
Hi @julka I don't really know what to do with this. Did you use it in a situation with AddThis support?
The implementation of addthis_plugin_info might be to close to a definition of a hook_plugin_info(). If I would merge it it needs to be something more abstract focused on the addthis situation.

Could you give a use-case for this functionality for me to consider better to merge it or not.

**Comment by julka** [comment]
Hi @matglas. 

When a user reports an issue to us with their website, our support team visits their site and uses a bookmarklet that analyzes the page and give them information that's helpful for troubleshooting. That bookmarklet uses a variety of information, including the global scope JavaScript variables addthis_config, addthis_share, and addthis_plugin_info. For instance, with the addthis_plugin_info, it can prompt them to use trouble shooting steps for a specific CMS or alert them if the site is using an outdated version of the module.

**Comment by julka** [comment]
By "our", I mean AddThis. I hadn't updated my employer on my github profile until just now.

Hi :-)

**Comment by matglas** [comment]
Aha. Great. Good to know :). There are a few things that i think need to be different in this approach. 

We need a setting to disable this. Because not every company likes there backend information go public for security reasons. Espacially the Drupal version that is included. And if its disabled the undefined check in the js needs to go.

The method with plugin info is inside the scriptmanager so I think thats ok.

It would be nice to include a notice in the readme about this information beeing 'published' or 'made public'.

**Comment by julka** [comment]
@matglas, I thought of another approach and pushed those changes. What do you think of this? It adds the checkbox you suggested, and depending on the value in the checkbox, it'll populate much less data into addthis_plugin_info. Basically, if it's unchecked, it only sets addthis_plugin_info.info_status to false, and nothing else. We can use this at AddThis to help indicate to our support team that we probably have troubleshooting steps for this particular plugin and that they can ask the user to check that box temporarily to help them work through an issue.

File: addthis.install [hunk 1]
```diff
@@ -99,10 +100,16 @@ function addthis_update_7402() {
...
 function addthis_update_7403() {
   variable_del('addthis_twitter_template');
 }
+
+/**
+ * Remove old PLUGIN_INFO_ENABLED_KEY variable.
+ */
+function addthis_update_7404() {
+  variable_set(AddThis::PLUGIN_INFO_ENABLED_KEY, TRUE);
+}
```
Comment (matglas) [inline addthis.install:115]: I would not force existing users to enable this. Only new users should have this as a default together with a notice in the README about this information beeing exposed.

File: classes/Services/AddThisScriptManager.php [hunk 2]
```diff
@@ -164,6 +165,28 @@ function setAsync($enabled) {
...
+   * Get a array with all addthis_plugin_info values.
+   */
+  private function getJsAddThisPluginInfo() {
+    if ($this->addthis->isPluginInfoEnabled()) {
+      $plugin_info = array(
+        'info_status' => 'enabled',
+        'cms_name' => 'Drupal',
+        'cms_version' => VERSION,
+        'plugin_name' => 'AddThis',
+        'plugin_version' => '7.x-4.0-alpha7',
```
Comment (matglas) [inline classes/Services/AddThisScriptManager.php]: You can get the current module version through the system_get_info() 

https://api.drupal.org/api/drupal/modules%21system%21system.module/function/system_get_info/7

**Comment by matglas** [comment]
I like the new approach better. I have some extra tweaks but I think its almost finished.

**Comment by julka** [comment]
I pushed the change to the upgrade default. I'm working on using system_get_info. Yay for not hard coding things! :-)

I was reading through the readme. Do you have any ideas on where and how you would want this presented there? There's no area that goes into detail about any settings. Would the configuration doc be a better location for that information? https://www.drupal.org/node/1309922

**Comment by julka** [comment]
Updated to use system_get_info

File: addthis.install [hunk 3]
```diff
@@ -99,10 +100,16 @@ function addthis_update_7402() {
...
- * TWITTER_TEXT_KEY
+/** 
+ * Remove old TWITTER_TEMPLATE_KEY variable
  */
 function addthis_update_7403() {
   variable_del('addthis_twitter_template');
 }
+
+/**
+ * Remove old PLUGIN_INFO_ENABLED_KEY variable.
```
Comment (matglas) [inline addthis.install]: I think this func doc is copied and needs some adjustment ;)

**Comment by matglas** [comment]
I do think the notice should be in the readme. Just make a new section with notices. 

When thats finished i think its good to create a issue on drupal.org and let the community have a last review.

**Comment by matglas** [comment]
Hi @julka did you see my request here about the change and the d.o. issue?

**Comment by julka** [comment]
Hi @matglas. Yes, I did. :-) I'm sorry, I've been very distracted this last week with other project. I will try to get to this soon.

**Comment by julka** [comment]
I created a d.o. issue here: https://www.drupal.org/node/2528512

**Comment by julka** [comment]
Added a notices section to the readme

**Pull request closed without merging**